### PR TITLE
Adding variable to define Custom browser to use

### PR DESCRIPTION
**Pull request by shubham-cpp**
Add an option for user to specify using different browser(other than default one) for org/markdown previews. Its working for me right now, but will need further  testing for any edge cases(if they exist).

Also updated README to include how to set the variable.

### CODE REVIEW

File: grip-mode.el [hunk 1]
```diff
@@ -98,6 +103,16 @@ option."
 (defvar-local grip--preview-file nil
   "The preview file for grip process.")
 
+(defun grip--browser (url)
+  "Use browser specified by user to load URL.
+Use default browser if nil."
+  (if grip-url-browser
+      (progn
+       (setq-local browse-url-generic-program grip-url-browser
```
Comment (seagle0128) [inline grip-mode.el]: Doesn't `let` work here? I prefer `let` than `setq-local`.

Or use this
```elisp
(when grip-url-browser
  (setq-local browse-url-generic-program grip-url-browser
              browse-url-browser-function 'browse-url-generic))
(browse-url url)
```

File: grip-mode.el [hunk 2]
```diff
@@ -56,6 +56,12 @@
   :type 'file
   :group 'grip)
 
+(defcustom grip-url-browser nil
+  "Browser to launch Markdown/Org previews.
+Use default browser if nil."
+  :type 'string
```
Comment (seagle0128) [inline grip-mode.el]: Should be: `:type '(choice (const :tag "None" nil) string)`.

**Review by seagle0128 — CHANGES_REQUESTED** [review]

File: grip-mode.el (hunk 2 above)
Comment (shubham-cpp) [reply grip-mode.el]: Ok that makes sense

File: grip-mode.el (hunk 1 above)
Comment (shubham-cpp) [reply grip-mode.el]: > let


For some reason let is causing issues with void function. **when** works as expected

File: grip-mode.el [hunk 3]
```diff
@@ -56,6 +56,12 @@
   :type 'file
   :group 'grip)
 
+(defcustom grip-url-browser nil
+  "Browser to launch Markdown/Org previews.
+Use default browser if nil."
+  :type '(choice (const :tag "None" nil) string)
+  :group 'grip)
+
```
Comment (seagle0128) [inline grip-mode.el:64]: I checked `browse-url.el` today, and I think another option is necessary.

```elisp
(defcustom grip-url-args nil
  "A list of strings defining options for `grip-url-browser'."
  :type '(repeat (string :tag "Argument")))
```

File: grip-mode.el [hunk 4]
```diff
@@ -98,6 +103,14 @@ option."
 (defvar-local grip--preview-file nil
   "The preview file for grip process.")
 
+(defun grip--browser (url)
+  "Use browser specified by user to load URL.
+Use default browser if nil."
+  (when grip-url-browser
+    (setq-local browse-url-generic-program grip-url-browser
+                browse-url-browser-function 'browse-url-generic))
```
Comment (seagle0128) [inline grip-mode.el]: Also set `grip-url-args`.

**Review by seagle0128 — CHANGES_REQUESTED** [review]
Overall it looks good now. Please make the changes, and then I will approve and merge. Thanks!

File: grip-mode.el (hunk 1 above)
Comment (seagle0128) [reply grip-mode.el]: OK.

File: grip-mode.el (hunk 4 above)
Comment (shubham-cpp) [reply grip-mode.el]: will look into it

Comment (seagle0128) [reply grip-mode.el]: Just use

```elisp
(setq-local browse-url-generic-program grip-url-browser
            browse-url-generic-args grip-url-args
            browse-url-browser-function 'browse-url-generic)
```

Comment (shubham-cpp) [reply grip-mode.el]: But what if user wants to pass arguments to default browser. I also though just add it to setq-local, but then it came to mind that maybe user wants to pass args to regular browser as well

Comment (seagle0128) [reply grip-mode.el]: So the best solution should be use `let`. The local vars won't be override. 
I tried this snippet and it works as expected.

```elisp
(defun grip--browser (url)
  "Use browser specified by user to load URL.
Use default browser if nil."
  (if grip-url-browser
      (let ((browse-url-generic-program grip-url-browser)
            (browse-url-generic-args grip-url-args))
        (ignore browse-url-generic-program)
        (ignore browse-url-generic-args)
        (browse-url-generic url))
    (browse-url url)))
```

Comment (shubham-cpp) [reply grip-mode.el]: yes. tested it. works as expected

**Comment by seagle0128** [comment]
Thank you!
